### PR TITLE
Add pathExists()

### PR DESCRIPTION
**Pull request by RyanZim**
This is my approach at resolving https://github.com/jprichardson/node-fs-extra/issues/145.

`fs.exists` is broken. We're a drop-in replacement for `fs`, so we can't fix it.

We can make a new method that acts like `fs.exists` should have behaved all along. I've named it `pathExists()`.

I also aliased `fs.existsSync` to `fs.pathExistsSync` for consistency.

## Example Usage
```js
const fs = require('fs-extra')

const file = '/tmp/this/path/does/not/exist/file.txt'
// Promise usage:
fs.pathExists(file)
  .then(exists => console.log(exists)) // => false
// Callback usage:
fs.pathExists(file, (err, exists) => {
  console.log(err) // => null
  console.log(exists) // => false
})
```

### CODE REVIEW

**Comment by manidlou** [comment]
LGTM! :+1:

**Comment by jprichardson** [comment]
I like it, nice work @RyanZim!

**Comment by jaredallard** [comment]
I'm interested in why `fs-extra` doesn't support using the not-broken `fs.exists` when accessing it via a promise. Seems like you could make it drop-in compatible via the promise version and keep using the broken version when NOT accessing it via a promise. Maybe I'm missing something though

**Comment by RyanZim** [comment]
@jaredallard We do promisify `fs.exists`, and still offer the broken `fs.exists` to callback users. However, we encourage the use of `fs.pathExists`, since it doesn't use deprecated Node APIs, and the callback version behaves correctly.

**Comment by jaredallard** [comment]
@RyanZim Yes, I know `fs.exists` is promisified, but I'm wondering why not have it point to/use the solution that `pathExists` implements, since you can literally rewrite `fs.exists -> fs.pathExists` right now. Since how `fs.exists` behaves (in it's broken way) is somewhat how it's consumed as a promise (`catch` is used as the `err` instead). 

**TL;DR**
The normal callback way _should_ point to the broken one for compat, but the `fs.exists` _promise_ version could be the same as the current `fs.pathExists` AFAIK.
